File: backend/PHP/MockProject-app/app/Models/Resident.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Resident extends Model
{
  use HasFactory;

  protected $table = 'Resident'; // Table name
  
  public $timestamps = false;     // Disable timestamps

  // Fillable properties
  protected $fillable = [
    'name',
    'birth_date',
    'email',
    'phone_number',
    'avatar',
    'username',
    'password',
    'role_id',
  ];
}

File: backend/apartment1/src/main/java/com/apartment/entity/Resident.java
package com.apartment.entity;

import jakarta.persistence.*;

import java.sql.Date;

@Entity
@Table(name = "Resident")
public class Resident {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(length = 100, nullable = false)
    private String name;

    private Date birthDate;

    @Column(length = 100)
    private String email;

    @Column(length = 20)
    private String phoneNumber;

    @Column(length = 255)
    private String avatar;

    @Column(length = 100, nullable = false)
    private String username;

    @Column(length = 100, nullable = false)
    private String password;

    @ManyToOne
    @JoinColumn(name = "role_id", referencedColumnName = "id")
    private Roles role;

}
